feature_engine/creation/distance_features.py

```diff
@@ -0,0 +1,241 @@
+        for idx, coordinate_column in enumerate(columns):
+            if len(coordinate_column) != 4:
+                raise ValueError(
+                    f"Needed 4 values to compute a distance, "
+                    f"but got {len(coordinate_column)} columns \n"
+                    f"at the index {idx} of the list coordinate columns."
+                )
+        return (
+            [coordinate[0] for coordinate in columns],
+            [coordinate[1] for coordinate in columns],
+            [coordinate[2] for coordinate in columns],
+            [coordinate[3] for coordinate in columns],
+        )
+
+    def _check_output_columns_names(
+        self,
+        column_name: Optional[List[Union[str, None]]],
+        coordinate_columns: List[List[Union[str, int]]],
+    ) -> Optional[List[Union[str, None]]]:
+        if column_name is None:
+            return [f"distance_{c[0]}_{c[1]}_{c[2]}_{c[3]}" for c in coordinate_columns]
+        if len(column_name) != len(coordinate_columns):
+            raise ValueError(
+                "Not enough output column names provided.\n "
+                f"Expected {len(coordinate_columns)} column names, "
+                f"but got {len(column_name)}."
+            )
+        return column_name
+
+    def fit(self, X: pd.DataFrame):
+        # there is no fit for this transformer
+        super().fit(X)
+        return self
+
+    def transform(self, X: pd.DataFrame):
+        """
+        Compute the distance on heart using the Haversine formula.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shame = [n_samples, n_features]
+            The data to be transformed.
+
+        Returns
+        -------
+        X_new: Pandas dataframe.
+            The original dataframe plus the distances between the given coordinates.
+        """
+        X = self._check_transform_input_and_state(X)
+        X = self._check_lat_lon_columns_are_in_df(X)
+        X = self._check_correctness_of_coordinates(X)
+
+        self._compute_distance(X)
+
+        if self.drop_original:
+            X.drop(
+                columns=[
+                    *self.a_latitudes,
+                    *self.a_longitudes,
+                    *self.b_latitudes,
+                    *self.b_longitudes,
+                ],
+                inplace=True,
+            )
+
+        return X
+
+    def _check_lat_lon_columns_are_in_df(self, X) -> pd.DataFrame:
+        coordinate_columns = [
+            *self.a_latitudes,
+            *self.a_longitudes,
+            *self.b_latitudes,
+            *self.b_longitudes,
+        ]
+        if set(coordinate_columns).issubset(set(X.columns)) is False:
+            raise ValueError(
+                f"The columns {set(coordinate_columns).issubset(set(X.columns))} "
+                f"were not found in the dataframe."
+            )
+        return X
+
+    def _check_correctness_of_coordinates(self, X: pd.DataFrame) -> pd.DataFrame:
```

Comment:
we need this check also in fit to control the input variables in the training set.